lasagne/layers/dnn.py

```diff
@@ -9,15 +8,16 @@
 from .pool import pool_output_length
 from ..utils import as_tuple
 
-if not theano.sandbox.cuda.cuda_enabled:
+if theano.sandbox.cuda.dnn.dnn_available():
```

Comment:
When device=cpu, you cannot access `theano.sandbox.cuda` if it's not explicitly imported. So this would need to be done differently.